GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java

```diff
@@ -529,80 +532,96 @@ private void analyseData(final List<InstanceData> allData) {
-        writeMetadata(sheet, instanceData, startRow, sheet.getLastRowNum() - startRow + 1,
-                showRepeatColumn);
     }
 
-    private void writeMetadata(Sheet sheet,
+
+    //We write one row at a time to stay in window
+    private void writeMetadataRow(Row r,
             InstanceData instanceData,
-            final int firstRowNo,
-            final int iterations,
+            final int iteration,
             final boolean showRepeatColumn) {
-        for (int i = 0; i < iterations; i++) {
-
-            Row r = getRow(firstRowNo + i, sheet);
-            SurveyInstanceDto dto = instanceData.surveyInstanceDto;
-            int col = 0;
-            // Write the identifier
-            createCell(r, col++, dto.getSurveyedLocaleIdentifier());
-            // Write data approval status
-            if (hasDataApproval()) {
-                createCell(r, col++, instanceData.latestApprovalStatus);
-            }
-            // Write the "Repeat" column
-            if (showRepeatColumn) {
-                createCell(r, col++, String.valueOf(i + 1), null, Cell.CELL_TYPE_NUMERIC);
-            }
-            // Write other metadata
-            createCell(r, col++, dto.getSurveyedLocaleDisplayName());
-            createCell(r, col++, dto.getDeviceIdentifier());
-            createCell(r, col++, dto.getKeyId().toString());
-            createCell(r, col++, ExportImportUtils.formatDateTime(dto.getCollectionDate()));
-            createCell(r, col++, sanitize(dto.getSubmitterName()));
-            String duration = getDurationText(dto.getSurveyalTime());
-            createCell(r, col++, duration);
-            Double v = dto.getFormVersion();
-            if (v != null) {	
-            	createCell(r, col++, v);
-            } 
-            //TODO else a cell with 0.0??
+        SurveyInstanceDto dto = instanceData.surveyInstanceDto;
+        int col = 0;
+        // Write the identifier
+        createCell(r, col++, dto.getSurveyedLocaleIdentifier());
+        // Write data approval status
+        if (hasDataApproval()) {
+            createCell(r, col++, instanceData.latestApprovalStatus);
         }
-
+        // Write the "Repeat" column
+        if (showRepeatColumn) {
+            createCell(r, col++, String.valueOf(iteration), null, Cell.CELL_TYPE_NUMERIC);
+        }
+        // Write other metadata
+        createCell(r, col++, dto.getSurveyedLocaleDisplayName());
+        createCell(r, col++, dto.getDeviceIdentifier());
+        createCell(r, col++, dto.getKeyId().toString());
+        createCell(r, col++, ExportImportUtils.formatDateTime(dto.getCollectionDate()));
+        createCell(r, col++, sanitize(dto.getSubmitterName()));
+        String duration = getDurationText(dto.getSurveyalTime());
+        createCell(r, col++, duration);
+        Double v = dto.getFormVersion();
+        if (v != null) {
+            createCell(r, col++, v);
+        }
+        //TODO else a cell with 0.0??
```

Comment:
Is this still needed?

Reply:
Might be. We have not had any feedback on the form version column.